hack/update-submodule.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Update submodules HEAD to remote branches
+
+function help() {
+    echo "Usage: update_submodules.sh [OPTIONS]... branch_name..."
+    echo "Update submodules HEAD to remote branches"
+    echo "Must be ran in repo root directory"
+    echo ""
+    echo "Options:"
+    echo "-h   Shows usage text"
+    echo "Example:"
+    echo "update_submodules.sh master release-4.6"
+}
+
+function update_branch_submodules() {
+    submodules=(kubernetes ovn-kubernetes windows-machine-config-bootstrapper windows_exporter containernetworking-plugins)
+    branch=$1
+    working_branch_name="$branch-SubmoduleUpdate$(date +%m-%d)"
+    top_dir=$(pwd)
+    echo Using branch $working_branch_name
+    git branch -D $working_branch_name
+    git checkout $branch -b $working_branch_name
+    git submodule update --init >/dev/null
+    for submodule in ${submodules[@]}; do
+        echo "--- Updating $submodule ---"
+        cd $top_dir
+        cd $submodule \
+        && origin_url=$(git remote get-url origin) \
+        && git fetch origin > /dev/null \
+        && git checkout origin/$branch \
+        && short_head=$(git rev-parse --short HEAD) \
+        && long_head=$(git rev-parse HEAD) \
+        && cd .. \
+        && git add $submodule \
```

Comment:
Can't we just do `git submodule update --remote --merge/rebasse` or something along those lines? This seems overly complicated.

Reply:
I also don't understand the need for `&&` instead of individual commands.